spring-cloud-gcp-data-firestore/src/main/java/org/springframework/cloud/gcp/data/firestore/FirestoreTemplate.java

```diff
@@ -194,30 +191,81 @@ public int getSaveAllBufferWriteSize() {
 		return findAllDocuments(entityClass, ID_PROJECTION).count();
 	}
 
+	/**
+	 * {@inheritdoc}
+	 *
+	 * <p>The buffer size and buffer timeout settings for {@link #saveAll} can be modified by calling
+	 * {@link #setWriteBufferSize} and {@link #setWriteBufferTimeout}.
+	 */
 	@Override
 	public <T> Mono<Long> deleteAll(Class<T> clazz) {
-		return findAllDocuments(clazz).map(Document::getName).flatMap(this::callDelete).count();
+		Flux<List<String>> documentBuffers =
+				findAllDocuments(clazz)
+						.map(Document::getName)
+						.bufferTimeout(this.writeBufferSize, this.writeBufferTimeout);
+
+		return ObservableReactiveUtil
+				.streamingBidirectionalCall(this::openWriteStream, documentBuffers, this::buildDeleteRequest)
+				.count();
 	}
 
+	/**
+	 * {@inheritdoc}
+	 *
+	 * <p>The buffer size and buffer timeout settings for {@link #saveAll} can be modified by calling
+	 * {@link #setWriteBufferSize} and {@link #setWriteBufferTimeout}.
+	 */
 	@Override
 	public <T> Mono<Void> delete(Publisher<T> entityPublisher) {
-		return Flux.from(entityPublisher).map(this::buildResourceName).flatMap(id -> callDelete(id)).then();
+		Flux<List<String>> documentBuffers =
+				Flux.from(entityPublisher)
+						.map(this::buildResourceName)
+						.bufferTimeout(this.writeBufferSize, this.writeBufferTimeout);
+
+		return ObservableReactiveUtil
+				.streamingBidirectionalCall(this::openWriteStream, documentBuffers, this::buildDeleteRequest)
+				.then();
 	}
 
+	/**
+	 * {@inheritdoc}
+	 *
+	 * <p>The buffer size and buffer timeout settings for {@link #saveAll} can be modified by calling
+	 * {@link #setWriteBufferSize} and {@link #setWriteBufferTimeout}.
+	 */
 	@Override
 	public <T> Mono<Void> deleteById(Publisher<String> idPublisher, Class entityClass) {
 		return Mono.defer(() -> {
-			FirestorePersistentEntity<?> persistentEntity = this.mappingContext.getPersistentEntity(entityClass);
-			return Flux.from(idPublisher).flatMap(id -> callDelete(buildResourceName(persistentEntity, id)))
+			FirestorePersistentEntity<?> persistentEntity =
+					this.mappingContext.getPersistentEntity(entityClass);
+
+			Flux<List<String>> documentBuffers =
+					Flux.from(idPublisher)
+							.map(id -> buildResourceName(persistentEntity, id))
+							.bufferTimeout(this.writeBufferSize, this.writeBufferTimeout);
```

Comment:
the following multiple lines are duplicated - maybe move that to a separate method?

Reply:
Done.